src/unstract/sdk/tool/base.py

```diff
@@ -253,8 +306,15 @@ def write_tool_result(self, data: Union[str, dict[str, Any]]) -> None:
         json_data = json.dumps(data)
         # INFILE is overwritten for next tool to run
         input_file_path: Path = Path(self.get_input_file())
-        with input_file_path.open("w", encoding="utf-8") as f:
-            f.write(json_data)
+        if self.workflow_filestorage:
+            ToolUtils.dump_json(
+                file_to_dump=input_file_path,
+                json_to_dump=data,
+                fs=self.workflow_filestorage,
+            )
+        else:
+            with input_file_path.open("w", encoding="utf-8") as f:
```

Comment:
Shouldn't this also be invoked on file storage instance? Let us discuss this. I am not able to clearly understand why the portions under else do not have an associated file storage instance.